src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java

```diff
@@ -542,10 +569,20 @@ public ListBoxModel doFillGroupLookupStrategyItems() {
             return model;
         }
 
-        public FormValidation doCheckRequireTLS() {
+        public FormValidation doCheckRequireTLS(@QueryParameter boolean requireTLS, @QueryParameter boolean startTls) {
+            if (isFipsNonCompliant(requireTLS, startTls)) {
+                return FormValidation.error(Messages.TlsConfiguration_ErrorMessage());
+            }
             Jenkins.get().checkPermission(Jenkins.ADMINISTER);
             if (System.getProperty(ActiveDirectoryAuthenticationProvider.ADSI_FLAGS_SYSTEM_PROPERTY_NAME) != null) {
-                return FormValidation.warning("This setting is overridden by the ADSI mode system property");
+                return FormValidation.error("This setting is overridden by the ADSI mode system property");
```

Comment:
```suggestion
                return FormValidation.warning("This setting is overridden by the ADSI mode system property");
```
Sorry, I think I made the suggestion of this change myself, but we should keep existing behaviour before this change, FIPS errors are errors, overrides should be warnings

Reply:
https://github.com/jenkinsci/active-directory-plugin/pull/184#discussion_r1696772071

Since this boolean value can be null, Boolean.valueOf will give NPE, which we will have to handle and then set the Boolean as FALSE in the catch. Therefore, I think using the ternary operator is better.